poetry/packages/project_package.py

```diff
@@ -8,6 +8,23 @@
 
 class ProjectPackage(Package):
     def __init__(self, name, version, pretty_version=None):
+        if isinstance(version, str) and version.startswith('attr:'):
```

Comment:
The __init__ code would look a little cleaner by moving this code into a new function.

Reply:
Sure, it's just a PoC.